ConversationHandler breaks when bot is also used in Channels

Rostgnom opened the pull request:
When using a `ConversationHandler` for private and group chats in conjunction to having the bot added to a Channel, it will throw a None type exception because it cannot retrieve a chat/user id from the update.

I suggest completely ignoring all channel_posts, because the interactivity that a ConversationHandler provides is not useful in channels anyway.

tsnoam commented on telegram/ext/conversationhandler.py:
```diff
@@ -116,6 +116,10 @@ def check_update(self, update):
             return False
 
+        # Ignore messages in channels
+        if update.channel_post:
```
Can you please "join" this with the `if` above?

Rostgnom commented:
Changed the file as requested. I also added a notion that the ConversationHandler currently does not work in a group chat with multiple users, in that the states do only switch for the user performing the transition/action. Is that clear from what I added to the class doc?
